src/api/goods.ts

```
@@ -4,13 +4,21 @@ import { Good } from '../types/Good';
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
 export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+  return fetch(API_URL)
+    .then(response => response.json())
+    .catch(error => new Error(error));
 }
 
 export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+  return getAll()
+    .then(goods =>
+      goods.sort((a, b) => a.name.localeCompare(b.name)).slice(0, 5),
+    )
+    .catch(error => new Error(error));
 };
 
 export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
+  return getAll()
+    .then(goods => goods.filter(good => good.color === 'red'))
+    .catch(error => new Error(error));
```

Similar to the previous functions, the error handling in `getRedGoods` can be improved. Instead of creating and returning a new `Error` object, consider re-throwing the error or returning a default value for better error management in the component.